src/formatDate.js

```diff
@@ -50,7 +50,74 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const arrDate = date.split(oldSeparator);
+
+  const result = [];
+  let year, month, day, oldYearFormat, newYearFormat;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case ('YY'):
+        oldYearFormat = 2;
+        year = arrDate[i];
+        break;
+
+      case ('YYYY'):
+        oldYearFormat = 4;
+        year = arrDate[i];
+        break;
+
+      case ('MM'):
+        month = arrDate[i];
+        break;
+
+      case ('DD'):
+        day = arrDate[i];
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    switch (toFormat[i]) {
+      case ('YY'):
+        newYearFormat = 2;
```

Comment:
why do you need `newYearFormat` variable?

Reply:
In one of the videos, it was said to limit the use of numeric data without variables, so that later we can know what the number is and where it came from. That's why I thought it would be better)